cfgov/preprocessed/js/modules/util/EventObserver.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+/**
+ * EventObserver
+ * @class
+ *
+ * @classdesc Used for creating an object
+ *   that can be used to dispatch and listen to custom events.
+ * @returns {Object} An EventObserver instance.
+ */
+function EventObserver() {
+
```

Comment:
> ok, I thought you might want to fire the event on the Dom Element. How do you access an Expandable instance from a Dom element?

That'd happen through the page scripts (in a global, template, or page-specific route).
So in example you posted, in `js/routes/common.js` there'd be something like:

```
function init() {

  // DOM references.
  var banner = new Expandable( '#beta-banner' ).init();

  // Event handlers.
  banner.addEventListener( 'beginExpand', _betaBannerClicked );

  // Initial state.
  if ( _storage.getItem( _key ) !== 'true' ) {
    banner.expand();
  }
}
```

The `Expandable.js` is composited on the DOM, but the DOM element itself is untouched. The rationale behind that is: 
- It would seem to break encapsulation. Through DOM querying, any particular DOM element is effectively global on a page, like `window`. It could be decorated with an `expand()` method but that doesn't safeguard against some other script adding a method of the same name to the element. 
- Finding the reference for an expandable doesn't require a DOM lookup. The only lookup that happens is at instantiation and from there all additional logic relies on events dispatched from the DOM.

That's the idea anyway! Happy to discuss why it may make sense to decorate the DOM element directly.

That example made me realize I need to add two things: Being able to initialize the expandable in an expanded state (currently that only happens when the CSS modifier is added) and broadcasting the click event.

> On a side-note, OAH has created something similar: https://github.com/cfpb/owning-a-home/blob/5567cf4d89c9ab65f327a98589ba9fd7dd93b2ee/src/static/js/modules/prepare-worksheets/util/event-observer.js.

It's similar because... I wrote that ;-)